gapic-generator-java/src/main/java/com/google/api/generator/gapic/protoparser/Parser.java

```diff
@@ -824,39 +824,34 @@ static LongrunningOperation parseLro(
     boolean isResponseTypeNameShortOnly = lastDotIndex < 0;
     String responseTypeShortName =
         lastDotIndex >= 0 ? responseTypeName.substring(lastDotIndex + 1) : responseTypeName;
+    // When only shortname is provided, match on same proto package as method (See
+    // https://aip.dev/151)
+    String responseTypeFullName =
+        isResponseTypeNameShortOnly
+            ? methodDescriptor.getFile().getPackage() + "." + responseTypeShortName
+            : responseTypeName;
```

Comment:
nit: Maybe we could refactor L823 - L843 to have a private helper method that checks for if the response/ metadata name is a shortname and returns the valid full type name back?

i.e.
```
String responseTypeFullName = getTypeFullName(packageName, responseTypeName);
String metadataTypeFullName = getTypeFullName(packageName, metadataTypeName);
```

and the helper func has the logic for checking the dot index, if it's a shortname, and concating the full type name back?

Reply:
Ah you're right - the code is doing very similar things for response and metadata type here. I'll take a look at refactoring!

Reply:
Updated in https://github.com/googleapis/sdk-platform-java/pull/1726/commits/de5e9c39c0d757200bddba4878c533f3104a8df3.